### PR TITLE
Find course show page

### PR DESCRIPTION
**Pull request by avinhurry**
### Context

This PR ports across the show page for the new Find. This includes several view components and the introduction of view previews for these components.

I've also brought preview on Publish in line with the new Find in this PR. 

### Changes proposed in this pull request

- Added a temporary course to the view so that we can easily view the page, until we have completed the rest of the search flow.
- Port across the view components and specs required for this page
- Add view previews and extra specs 
- See commits for more details

### Guidance to review

- Navigate to: https://publish-teacher-training-pr-2966.london.cloudapps.digital/find/course/:provider_code/:course_code

- Try a variety of different courses, by changing the `course_code` and `name` at the top of `find/courses/show.html.erb`, ensure they all match what Find currently displays on production.
- Repeat for Preview. 

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by avinhurry** [comment]
[Course Show Page](https://trello.com/c/hANFXSMl/388-course-show-page)

File: app/components/find_interface/courses/about_schools_component/view.html.erb [hunk 1]
```diff
@@ -0,0 +1,53 @@
+<div class="govuk-!-margin-bottom-8">
+  <h2 class="govuk-heading-l" id="section-schools"><%= course.placements_heading %></h2>
+  <div data-qa="course__about_schools">
+    <% if course.program_type == "higher_education_programme" && course.provider.provider_code != "B31" %>
```
Comment (defong) [inline app/components/find_interface/courses/about_schools_component/view.html.erb:4]: Can we stick this in a helper with a tests please

ie
```ruby
course.higher_education_programme?
```


`B31` looks vile, if that no longer exist can we remove please

File: app/components/find_interface/courses/about_schools_component/view.html.erb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      <%= render FindInterface::Utility::AdviceComponent::View.new(title: "Where you will train") do %>
+        <p class="govuk-body">You’ll be placed in schools for most of your course. Your school placements will be within
+          commuting distance.</p>
+        <p class="govuk-body">You can’t pick which schools you want to be in, but your university will try to take your
+          journey time into consideration.</p>
+        <p class="govuk-body">Universities can work with over 100 potential placement schools. Most will be within 10
+          miles of the university, but sometimes they can cover a wider area, especially outside of cities.</p>
+      <% end %>
+    <% elsif course.program_type == "scitt_programme" && course.provider.provider_code != "E65" %>
```
Comment (defong) [inline app/components/find_interface/courses/about_schools_component/view.html.erb:13]: Can we stick this in a helper with a tests please
ie

```ruby

course.scitt_programme?

```

`E65` looks vile, if that no longer exist can we remove please

File: app/components/find_interface/courses/about_schools_component/view.html.erb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      <%= render FindInterface::Utility::AdviceComponent::View.new(title: "Where you will train") do %>
+        <p class="govuk-body">You’ll be placed in different schools during your training. You can’t pick which schools
+          you want to be in, but your course will try to place you in schools you can commute to.</p>
+      <% end %>
```
Comment (defong) [inline app/components/find_interface/courses/about_schools_component/view.html.erb:17]: the block looks wrong

it should either be in the component (probably looks nicer) or

``` ruby

<%= render FindInterface::Utility::AdviceComponent::View.new(title: "Where you will train") do %>
     <% if course.program_type == "higher_education_programme" && course.provider.provider_code != "B31" %>
foo
     <% elsif course.program_type == "scitt_programme" && course.provider.provider_code != "E65" %>
bar
     <% else %>
foobar
<% end %>

     ```

File: app/components/find_interface/courses/contact_details_component/view.html.erb [hunk 4]
```diff
@@ -3,28 +3,28 @@
 
   <div class="course-basicinfo">
     <dl class="app-description-list">
-      <% if @provider.email.present? %>
+      <% if course.provider.email.present? %>
```
Comment (defong) [inline app/components/find_interface/courses/contact_details_component/view.html.erb:6]:
```suggestion
      <% if provider.email.present? %>
```

It looks like you have delegate in place

File: app/components/find_interface/courses/contact_details_component/view.rb [hunk 5]
```diff
@@ -0,0 +1,14 @@
+module FindInterface
+  module Courses
+    class ContactDetailsComponent::View < ViewComponent::Base
+      attr_reader :course
+
+      delegate :provider, to: :course
```
Comment (defong) [inline app/components/find_interface/courses/contact_details_component/view.rb:6]: this delegate

File: app/components/find_interface/courses/about_schools_component/view.html.erb [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    <% end %>
+
+    <% if course.how_school_placements_work.present? %>
+      <%= markdown(course.how_school_placements_work) %>
+    <% end %>
+
+    <% if course.site_statuses.map(&:site).uniq.many? %>
```
Comment (defong) [inline app/components/find_interface/courses/about_schools_component/view.html.erb:24]: Should be view component concern not here

File: app/components/find_interface/courses/contents_component/view.html.erb [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+  <% if interview_process.present? %>
+    <li><%= govuk_link_to "Interview process", "#section-interviews" %></li>
+  <% end %>
+  <li><%= govuk_link_to "International candidates", "#section-international-students" %></li>
+  <% if provider.train_with_disability.present? %>
+    <li><%= govuk_link_to "Training with disabilities and other needs", "#section-train-with-disabilities" %></li>
+  <% end %>
+  <li><%= govuk_link_to "Contact this training provider", "#section-contact" %></li>
+  <li><%= govuk_link_to "Support and advice", "#section-advice" %></li>
+  <% if course.has_vacancies? %> <li><%= govuk_link_to "Apply", "#section-apply" %></li> <% end %>
```
Comment (defong) [inline app/components/find_interface/courses/contents_component/view.html.erb:26]: This spaghetti logic should be in view component....

...this looks like the worse code reading exercise ever

**Review by defong — APPROVED** [review]
Discussed offline, this is a copy and pasta job, hence all tech debt is beyond scope

**Pull request merged**